crates/electrum/src/electrum_ext.rs

```diff
@@ -231,35 +214,20 @@ impl ElectrumExt<ConfirmationHeightAnchor> for Client {
                 .map(|cp| (cp.height(), cp))
                 .collect::<BTreeMap<u32, CheckPoint>>();
 
-            if !request_spks.is_empty() {
-                if !scanned_spks.is_empty() {
-                    scanned_spks.append(&mut populate_with_spks(
+            for (keychain, keychain_spks) in &mut request_spks {
+                scanned_spks.extend(
+                    populate_with_spks(
                         self,
                         &cps,
                         &mut update,
-                        &mut scanned_spks
-                            .iter()
-                            .map(|(i, (spk, _))| (i.clone(), spk.clone())),
-                        stop_gap,
-                        batch_size,
-                    )?);
-                }
-                for (keychain, keychain_spks) in &mut request_spks {
-                    scanned_spks.extend(
-                        populate_with_spks(
-                            self,
-                            &cps,
-                            &mut update,
-                            keychain_spks,
-                            stop_gap,
-                            batch_size,
-                        )?
-                        .into_iter()
-                        .map(|(spk_i, spk)| ((keychain.clone(), spk_i), spk)),
-                    );
-                }
+                        keychain_spks,
+                        usize::default(),
+                        usize::default(),
```

Comment:
I'm pretty sure we should keep `stop_gap` and `batch_size` as arguments. Using `usize::default()` instead produces the wrong result.